### DISCUSSION TITLE
Is there an abstract interface for online video sites? 是否有针对在线视频网站的抽象接口？

### DISCUSSION THREAD

**Discussion by bcdax110** [aidenlx/media-extended#341] [Ideas]
插件中有没有一种通用接口，针对不同的视频网站只要实现了这个接口，就可以使插件支持？这样在支持更多的视频网站方面会变得更容易。比如有人想支持某个网站，且也有一定的编码能力，就不需要研究插件中其他代码的作用，直接实现这个接口提个PR就可以。
---
Is there a common interface in the plugin for different video sites that would allow the plugin to support them as long as that interface is implemented? This would make it easier in terms of supporting more video sites. For example, if someone wants to support a certain site and also has some coding skills, they don't need to study the role of other code in the plugin, they can just implement this interface and mention a PR.

**Comment by aidenlx** [comment]
这块不用担心，其实设计当初就已经是按插件写的了，没有耦合很深，主要是我没法腾出精力写开发相关的教程
